### PR TITLE
Implement account categories

### PR DESCRIPTION
**Pull request by unnawut**
Issue/Task Number: T275

# Overview

This PR adds the ability to tag an account with categories.

# Changes

- Add swagger specs for `/category.all`, `/category.get`, `/category.create`, `/category.update`, `/category.delete`
- Add migrations for the category
- Add Category schema
- Add related routes, controllers, views, serializers for endpoints above
- Add category into the sample seed
- Add `Account.add_category/1` and `Account.remove_category/2`

# Implementation Details

Straightforward many-to-many associations between accounts and categories.

# Usage

Try out endpoints in the Category section in `/admin/api/docs`

# Impact

Deploy as usual.

### CODE REVIEW

File: apps/ewallet/lib/ewallet/policies/category_policy.ex [hunk 1]
```diff
@@ -0,0 +1,25 @@
+defmodule EWallet.CategoryPolicy do
+  @moduledoc """
+  The authorization policy for accounts.
```
Comment (T-Dnzt) [inline apps/ewallet/lib/ewallet/policies/category_policy.ex]: for categories

File: apps/admin_api/lib/admin_api/v1/controllers/category_controller.ex [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  """
+  def all(conn, attrs) do
+    with :ok <- permit(:all, conn.assigns.user.id, nil) do
+      categories =
+        Category
+        |> SearchParser.to_query(attrs, @search_fields, @mapped_fields)
+        |> SortParser.to_query(attrs, @sort_fields, @mapped_fields)
+        |> Paginator.paginate_attrs(attrs)
+
+      case categories do
```
Comment (T-Dnzt) [inline apps/admin_api/lib/admin_api/v1/controllers/category_controller.ex]: No `respond_multiple` / `respond_single`? 🤔

File: apps/ewallet/lib/ewallet/web/v1/serializers/account_serializer.ex [hunk 3]
```diff
@@ -31,6 +31,7 @@ defmodule EWallet.Web.V1.AccountSerializer do
       name: account.name,
       description: account.description,
       master: Account.master?(account),
+      category_ids: CategorySerializer.serialize(account.categories, :id),
```
Comment (T-Dnzt) [inline apps/ewallet/lib/ewallet/web/v1/serializers/account_serializer.ex]: No `categories` field that can be preloaded to send back all the categories? For now, I don't see how we can get all the categories for an account 🤔

File: apps/admin_api/lib/admin_api/v1/controllers/category_controller.ex (hunk 2 above)
Comment (unnawut) [reply apps/admin_api/lib/admin_api/v1/controllers/category_controller.ex]: The controllers started to have branching and the `respond_multiple` / `respond_single` are difficult to handle now.

Will do a separate PR that creates macro like `if_allowed user, :get, object_id, action do ... end` instead.

**Comment by unnawut** [comment]
@T-Dnzt Please check again

**Pull request merged**